text/0000-near-did-method.md

```diff
@@ -0,0 +1,170 @@
+
+This specification conforms to the requirements specified in the [DIDs specification](https://www.w3.org/TR/did-core/) currently published by the W3C Credentials Community Group.
+
+# Motivation
+[motivation]: #motivation
+
+NEAR uses readable account identifiers instead of a hash of a public key, and the accounts have some DID features, but not all. We have developed this specification to define a new DID method for hosting DIDs on the NEAR blockchain, also referred to as NEAR DID, and facilitate developers to work with related contracts.
+
+# NEAR DID Method Name
+[near-did-method-name]: #near-did-method-name
+
+The namestring that shall identify this DID method is: `near`.
+
+A DID that uses this method **MUST** begin with the following prefix: `did:near`. Per this DID specification, this string **MUST** be in lowercase.
+
+The remainder of the DID, after the prefix, is its namespace-specific identifier specified below.
+
+## Namespace Specific Identifier (NSI)
+[namespace-specific-identifier]: #namespace-specific-identifier
+
+The namespace specific identifier of NEAR DID is defined by the following ABNF:
+```
+near-did   = "did:near:" near-specific-idstring
+near-specific-idstring = *( 1*idchar ".") 2*idchar
+idchar    = %x30-39 / %x61-7A / "-" / "_"
+```
+The maximum length of `near-specific-idstring` is 64.
+
+## Example
+[example]: #example
+
+A valid NEAR DID might be:
+```
+did:near:alice.near
+```
+
+# CRUD Operations
+[crud-operations]: #crud-operations
+
+The following section outlines the DID operations for the `did:near` method.
+
+NEAR DIDs reside on the NEAR blockchain, and are mananged via the NEAR DID management smart contract.
+
+For the sake of convenience, we refer to the NEAR DID management smart contract as 'the registry'.
+
+## Create (Register)
+[create]: #create
+
+A Near account is needed to create a new NEAR DID,
+
+NEAR DID creation is done by sending a transaction to the registry invoking the following method:
+
+```
+reg_did_using_account()
+```
+
+Upon invoking this method, a new NEAR DID is generated by concatenating "did:near:" and the NEAR account identifier.
+
+The public key of the Near account's access key that was used to sign the transaction will be inserted into the `publicKey` property in the valid form and will be referred to in the `authentication` property.
+
+This method will return true if the new NEAR DID is registered successfully.
+
+## Read (Resolve)
+[read]: #read
+
+NEAR DID's associated DID document can be looked up by invoking the `getDocument` method of the registry.
+
+To ensure the smart contract invocation result is trustworthy, the client could query a certain number of nodes and then compare the return values.

```

Comment:
I don't think it's necessary. If the client doesn't trust some public rpc server they can use their own.

Reply:
> I don't think it's necessary. If the client doesn't trust some public rpc server they can use their own.

Sure, this definitely isn't a necessity, just a possible solution. Clients may or may not run their own server. It is up to them to use the server of their choice.